### PR TITLE
Add Git cli calls into output channel

### PR DESCRIPTION
**Pull request by mattseddon**
# 1/2 `main` <- this <- #2226

This PR moves all calls made to Git into the DVC output channel. This will make the user aware of these calls and should let them debug should something go wrong.

### Demo

https://user-images.githubusercontent.com/37993418/185831174-2e480fb5-f36f-40f3-9116-d0559cc5148d.mov

### CODE REVIEW

File: extension/src/git.ts [hunk 1]
```diff
@@ -10,98 +8,9 @@ export const GIT_REFS = join(DOT_GIT, 'refs')
 export const GIT_LOGS_REFS = join(DOT_GIT, 'logs', 'refs')
 export const HEADS_GIT_REFS = join(GIT_REFS, 'heads')
 
-const getUris = (repositoryRoot: string, relativePaths: string[]) =>
```
Comment (mattseddon) [inline extension/src/git.ts:13]: [F] These are the all the functions that got moved into the new classes.

File: extension/src/repository/commands/register.ts [hunk 2]
```diff
@@ -37,14 +37,14 @@ const registerResourceGroupCommands = (
   repositories: WorkspaceRepositories,
   internalCommands: InternalCommands
 ) => {
-  internalCommands.registerExternalCommand<Root>(
-    RegisteredCommands.GIT_STAGE_ALL,
-    getStageAllCommand(repositories)
+  internalCommands.registerExternalCliCommand<Root>(
+    RegisteredCliCommands.GIT_STAGE_ALL,
+    getStageAllCommand(repositories, internalCommands)
```
Comment (mattseddon) [inline extension/src/repository/commands/register.ts:42]: [F] The extension will now "offer to show errors" (open the output channel) if these commands fail.

**Comment by mattseddon** [comment]
Will have to integrate with #2091

File: extension/src/test/suite/repository/sourceControlManagement.test.ts [hunk 3]
```diff
@@ -262,12 +266,14 @@ suite('Source Control Management Test Suite', () => {
...
       expect(mockCheckout).not.to.be.called
-      expect(mockGitReset).not.to.be.called
+      expect(mockExecuteProcess).not.to.be.called
     })
 
     it('should reset the workspace if the user confirms they want to', async () => {
       const mockCheckout = stub(CliExecutor.prototype, 'checkout').resolves('')
-      const mockGitReset = stub(ProcessExecution, 'executeProcess').resolves('')
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const mockExecuteProcess = stub(Cli.prototype as any, 'executeProcess')
```
Comment (sroy3) [inline extension/src/test/suite/repository/sourceControlManagement.test.ts:275]: Move to a util so that we only have one `elint-disable`?

**Review by sroy3 — APPROVED** [review]

File: extension/src/test/suite/repository/sourceControlManagement.test.ts (hunk 3 above)
Comment (mattseddon) [reply extension/src/test/suite/repository/sourceControlManagement.test.ts:275]: I'll sort this out for good by adding a test util as a follow-up today.

**Pull request merged**